### PR TITLE
Second round of refactoring for Manager

### PR DESCRIPTION
**Pull request by jackgr**
/cc @sparkprime, @michelleN 

This is a large refactoring that builds and tests, but doesn't expand charts, yet. Next pass will include expansion changes to connect w/work done by @sparkprime on expandybird.

@michelleN I see your work in progress on ChartRepos, and left it for you to continue. However, if you prefer, I can finish it out in the next pass. In that case, would like to chat w/you about a couple of details.

### CODE REVIEW

**Comment by michelleN** [comment]
@jackgr Might be best if you want to finish it off since you're already on a roll with the refactor. I just stubbed those routes in so the client had something to talk to in the meantime. There might be some duplication in this PR with the intent of https://github.com/kubernetes/helm/blob/master/cmd/manager/chartrepos.go. 
Let's chat a bit about how we want to continue + details.

File: cmd/manager/deployments.go [hunk 1]
```diff
@@ -479,23 +489,23 @@ func getRegistry(w http.ResponseWriter, r *http.Request, handler string) *common
...
-		e := fmt.Errorf("Registry name does not match %s != %s", reg.Name, registryName)
+	reg := getRepo(w, r, handler)
+	if reg.Name != repoName {
+		e := fmt.Errorf("Repo name does not match %s != %s", reg.Name, repoName)
 		httputil.BadRequest(w, r, e)
 		return nil
 	}
 	if reg != nil {
-		err = c.Manager.CreateRegistry(reg)
+		err = c.Manager.CreateRepo(reg)
```
Comment (michelleN) [inline cmd/manager/deployments.go]: Is this different from AddChartRepo on cmd/manager/manager/manager.go?

**Comment by michelleN** [comment]
Looks good.
Next steps (as discussed) include..
- removing name from repo struct
- addChartRepo will take a repo struct where URL cannot be nil
- removeChartRepo will take a url instead of name
... etc.

File: cmd/manager/deployments.go (hunk 1 above)
Comment (jackgr) [inline cmd/manager/deployments.go]: It takes the chart as a payload, not just the URL. Per our discussion today, I will unify the two as AddChartRepo.

**Comment by jackgr** [comment]
Thanks for the review, @michelleN! It was a lot to look at.

**Pull request merged**